src/Datadog.Trace/Configuration/JsonConfigurationSource.cs

```diff
@@ -97,5 +99,40 @@ public T GetValue<T>(string key)
                        ? default
                        : token.Value<T>();
         }
+
+        /// <summary>
+        /// Gets a <see cref="ConcurrentDictionary{TKey, TValue}"/> containing all of the values.
+        /// </summary>
+        /// <remarks>
+        /// Example JSON where `globalTags` is the configuration key.
+        /// {
+        ///  "globalTags": {
+        ///     "name1": "value1",
+        ///     "name2": "value2"
+        ///     }
+        /// }
+        /// </remarks>
+        /// <param name="key">The key that identifies the setting.</param>
+        /// <returns><see cref="IDictionary{TKey, TValue}"/> containing all of the key-value pairs.</returns>
+        /// <exception cref="JsonReaderException">Thrown if the configuration value is not a valid JSON string.</exception>"
+        public IDictionary<string, string> GetDictionary(string key)
+        {
+            var token = _configuration.SelectToken(key, errorWhenNoMatch: false);
+            if (token == null)
+            {
+                return null;
+            }
+
+            // Presence of a curly brace indicates that this is likely a JSON string. An exception will be thrown
+            // if it fails to parse or convert to a dictionary.
+            if (token.Type == JTokenType.Object)
+            {
+                var dictionary = token
+                    ?.ToObject<ConcurrentDictionary<string, string>>() as ConcurrentDictionary<string, string>;
```

Comment:
nit: I think the `as ConcurrentDictionary<string,string>` is redundant with the parameterized `ToObject` call

Reply:
ah, yes! I just removed another of these earlier. Thanks for catching it!

Reply:
@zacharycmontoya I take that back, actually, the reason I put the `as ConcurrentDictionary...` in there is to avoid an exception being thrown. I think we need it there.

Reply:
We don't need the cast. The generic `ToObject<T>()` returns `T`, that cast is a no-op.

What exception were you expecting?

Reply:
If the cast fails at runtime, an `InvalidCastException` could be thrown. I don't know, maybe I'm wrong.

Reply:
Oh, I see. You're not wrong. We should probabl wrap this in a try/catch in case the json is not valid (i.e. neither a string or an object).

But note that if the call to `ToObject<T>()` itself fails and throws an exception, that will happen whether we have that cast or not.

Reply:
The `as <type>` construction is intended to avoid the exception. It will return `null` instead of throwing. Can't seem to find the docs on it at the moment, but check out https://www.geeksforgeeks.org/c-sharp-as-operator-keyword/